### PR TITLE
Update to gradle v9

### PR DESCRIPTION
**Pull request by jaydeluca**
Supersedes #7523

### CODE REVIEW

File: sdk-extensions/jaeger-remote-sampler/build.gradle.kts [hunk 1]
```diff
@@ -79,4 +79,8 @@ tasks {
       compilerArgs.add("-Xlint:-serial")
     }
   }
+
+  checkstyleMain {
+    source = fileTree("src/main/java")
+  }
```
Comment (trask) [inline sdk-extensions/jaeger-remote-sampler/build.gradle.kts]: why is this needed?

**Review by trask — APPROVED** [review]
Thanks!

File: sdk-extensions/jaeger-remote-sampler/build.gradle.kts (hunk 1 above)
Comment (jaydeluca) [reply sdk-extensions/jaeger-remote-sampler/build.gradle.kts]: the build was failing because it was [analyzing the generated protos](https://github.com/open-telemetry/opentelemetry-java/actions/runs/17049969193/job/48334958161) and generating errors like:

> Error: eckstyle] [ERROR] /Users/runner/work/opentelemetry-java/opentelemetry-java/sdk-extensions/jaeger-remote-sampler/build/generated/source/proto/main/java/io/opentelemetry/sdk/extension/trace/jaeger/proto/api_v2/Sampling.java:3885:7: 'if' construct must use '{}'s. [NeedBraces]

I wasn't able to track down why it started analyzing the protos after the gradle update, I'm wondering if the order of which the plugins are applied is different now.

Comment (trask) [reply sdk-extensions/jaeger-remote-sampler/build.gradle.kts]:
```suggestion

  checkstyleMain {
    // overrides the default which includes generated proto sources
    source = fileTree("src/main/java")
  }
```

**Review by breedx-splk — APPROVED** [review]
Amazing! Thanks @jaydeluca. Curious if you tried any of the publish work (like even `./gradlew publishToMavenLocal`)?

**Comment by jaydeluca** [comment]
@breedx-splk yes publishToMavenLocal looks good for me locally, are there others I would be able to test that you think would be useful?

**Comment by breedx-splk** [comment]
> @breedx-splk yes publishToMavenLocal looks good for me locally, are there others I would be able to test that you think would be useful?

Not off the top of my head -- just thinking about potential edge cases that don't get flexed in the normal PR and main merge builds. Thanks again!

**Review by jkwatson — APPROVED** [review]
Thanks!
